app/views/contexts/section_summary_context.py

```diff
@@ -236,3 +307,52 @@ def _get_safe_page_title(self, title):
         return (
             safe_content(self._schema.get_single_string_value(title)) if title else ""
         )
+
+    @staticmethod
+    def _get_add_or_edit_blocks_primary(
+        primary_person_block: Mapping[str, Any]
+    ) -> tuple[str, str]:
+        primary_person_edit_block_id = primary_person_block["add_or_edit_block"]["id"]
+        edit_block_id = primary_person_block["add_or_edit_block"]["id"]
+
+        return primary_person_edit_block_id, edit_block_id
+
+    def _get_related_answers(self, current_list: ListModel) -> dict[str, list]:
+        section = self.section["id"]
+
+        # pylint: disable=protected-access
+        if related_answers := self._schema._get_related_answers_for_section(section):
+            related_answers_dict = {}
+            for list_id in current_list:
+                keys = [
+                    self._schema.get_answers_by_answer_id(answer.answer_id)[0]["label"]
+                    for answer in self._answer_store
+                    if answer.answer_id in related_answers
+                    and answer.list_item_id == list_id
+                ]
+                values = [
+                    answer.value
+                    for answer in self._answer_store
+                    if answer.answer_id in related_answers
+                    and answer.list_item_id == list_id
+                ]
+
+                ids = [
+                    answer.answer_id
+                    for answer in self._answer_store
+                    if answer.answer_id in related_answers
+                    and answer.list_item_id == list_id
+                ]
+
+                related_answers_dict[list_id] = list(zip(keys, values, ids))
+
+            return related_answers_dict
+
+    @staticmethod
+    def _get_answer_title(list_collector_block: Mapping[str, Any]) -> str:
+        if variants := list_collector_block["add_block"].get("question_variants"):
+            if question := variants[0].get("question"):
+                return question["answers"][0]["label"]
+            return variants[0]["answers"][0]["label"]
```

Comment:
We can't just pick the first variant unless we enforce that the labels will be the same.

Reply:
We don't enforce it so I refactored it to use `choose_variant` method.